### PR TITLE
Fix Integer Clamp

### PR DESCRIPTION
**Pull request by cwfitzgerald**
**Connections**

Discovered during my work on #3159.

**Description**

`clamp(3, 2, 1)` for integers is explicitly defined by the [wgsl spec](https://gpuweb.github.io/gpuweb/wgsl/#clamp) as returning 1. In practice this happens on OpenGL on nvidia cards and on vulkan on AMD cards.

**Testing**

Have a PR coming up with tests for int clamp, as well as other shader functions

### CODE REVIEW

File: naga/src/back/glsl/mod.rs [hunk 1]
```diff
@@ -3138,7 +3138,29 @@ impl<'a, W: Write> Writer<'a, W> {
...
+                        match scalar_kind {
+                            crate::ScalarKind::Float => "clamp",
+                            // Clamp is undefined if min > max. In practice this means it can use a median-of-three
+                            // instruction to determine the value. This is fine according to the WGSL spec for float
+                            // clamp, but integer clamp _must_ use min-max. As such we write out min/max.
+                            _ => {
+                                write!(self.out, "min(max(")?;
+                                self.write_expr(arg, ctx)?;
+                                write!(self.out, ", ")?;
+                                self.write_expr(arg1.unwrap(), ctx)?;
```
Comment (Wumpf) [inline naga/src/back/glsl/mod.rs:3155]: Naga beginner question: can we know for sure at this point that `arg1` and `arg2` are `Some`, i.e. have we already made sure that the input is valid?

**Review by Wumpf — APPROVED** [review]
huh, interesting detail. Nothing is ever easy!

**Comment by Wumpf** [comment]
warrants a changelog entry imho

File: naga/src/back/glsl/mod.rs (hunk 1 above)
Comment (cwfitzgerald) [reply naga/src/back/glsl/mod.rs:3155]: Yeah, the Naga validator should have made sure that the argument count is correct

**Review by teoxoy — APPROVED** [review]

**Pull request merged**